mobile/mobile-core/devices-service.ts

```diff
@@ -245,21 +252,37 @@ export class DevicesService implements Mobile.IDevicesService {
 		}
 	}
 
+	/**
+	 * Returns device depending on the passed index.
+	 * The index refers to assigned number to listed devices by tns device command.
+	 * @param index assigned device number
+	 */
 	private getDeviceByIndex(index: number): Mobile.IDevice {
 		this.validateIndex(index - 1);
 		return this.getDeviceInstances()[index - 1];
 	}
 
+	/**
+	 * Returns running device for specified --device <DeviceId>.
+	 * Method expects running devices.
+	 * @param identifier parameter passed by the user to --device flag
+	 */
 	private async getDevice(deviceOption: string): Promise<Mobile.IDevice> {
 		await this.detectCurrentlyAttachedDevices();
 		let device: Mobile.IDevice = null;
 
-		if (this.hasDevice(deviceOption)) {
-			device = this.getDeviceByIdentifier(deviceOption);
+		let emulatorIdentifier = null;
+		if (this._platform) {
+			let emulatorService = this.resolveEmulatorServices();
+			emulatorIdentifier = await emulatorService.getRunningEmulatorId(deviceOption);
+		}
+
+		if (this.hasRunningDevice(emulatorIdentifier)) {
```

Comment:
We pass the emulatorIdentifier from `emulatorService.getRunningEmulatorId(). Try it out: if the case is, connected emulator and a connected device and you want to run on the emulator you can do the following:
tns run android --device <emulator index>
tns run android --device <emulator-5554/6 ...>
tns run android --device <emulator_image>
This will start the app only on the specified emulator.